src/lib/dialog/dialog-ref.ts

```diff
@@ -87,22 +90,33 @@ export class MatDialogRef<T, R = any> {
    * @param dialogResult Optional result to return to the dialog opener.
    */
   close(dialogResult?: R): void {
-    this._result = dialogResult;
+    let actualResult =  dialogResult ? dialogResult : this._defaultResult;
+    this._result = actualResult;
 
     // Transition the backdrop in parallel to the dialog.
     this._containerInstance._animationStateChanged.pipe(
       filter(event => event.phaseName === 'start'),
       take(1)
     )
     .subscribe(() => {
-      this._beforeClosed.next(dialogResult);
+      this._beforeClosed.next(actualResult);
       this._beforeClosed.complete();
       this._overlayRef.detachBackdrop();
     });
 
     this._containerInstance._startExitAnimation();
   }
 
+  /**
+   * Set a result that will be used as the default if no other result is provided.
+   * @param defaultResult Result that will be used as the default.
+   */
+  setDefaultResult(defaultResult: R): this {
```

Comment:
It's a little weird to have this method be on the `MatDialogRef` since it implies that the dialog itself is the one that knows what the default result should be, rather than the thing that opened the dialog. It might be more appropriate to have it on the `MatDialogConfig` instead.

Reply:
I actually could see both ways be implemented. The dialog is the one providing different results based on the user's actions so it was natural to me that the dialog should also be the one to define a default option. But I can also see use cases where one would like to define the default result from the `MatDialogConfig`.

And in the case that both options would be set I think the one from the `MatDialogConfig` should have precedence. What do you think?

Reply:
I updated my PR and added the option to set a default in the `MatDialogConfig`.